lib/android-tools-info.ts

```diff
@@ -92,7 +149,31 @@ export class AndroidToolsInfo implements IAndroidToolsInfo {
 				}
 			}
 
-			return detectedErrors;
+			return detectedErrors || isAndroidHomeValid;
+		}).future<boolean>()();
+	}
+
+	public validateJavacVersion(installedJavaVersion: string, options?: {showWarningsAsErrors: boolean}): IFuture<boolean> {
+		return ((): boolean => {
+			let hasProblemWithJavaVersion = false;
+			if(options) {
+				this.showWarningsAsErrors = options.showWarningsAsErrors;
+			}
+			let additionalMessage = "You will not be able to build your projects for Android." + EOL
+				+ "To be able to build for Android, verify that you have installed The Java Development Kit (JDK) and configured it according to system requirements as" + EOL +
+				" described in https://github.com/NativeScript/nativescript-cli#system-requirements.";
+			let matchingVersion = (installedJavaVersion || "").match(AndroidToolsInfo.VERSION_REGEX);
+			if(matchingVersion && matchingVersion[1]) {
+				if(semver.lt(matchingVersion[1], AndroidToolsInfo.MIN_JAVA_VERSION)) {
+					hasProblemWithJavaVersion = true;
+					this.printMessage(`Javac version ${installedJavaVersion} is not supported. You have to install at least ${AndroidToolsInfo.MIN_JAVA_VERSION}.`, additionalMessage);
+				}
+			} else {
+				hasProblemWithJavaVersion = true;
+				this.printMessage("Error executing command 'javac'. Make sure you have java installed and set JAVA_HOME environment variable.", additionalMessage);
```

Comment:
It is not enough to install java (the JRE). you need to install JDK. THis should be reflected in the message.